pkg/gen/input/workflows/internal/file/lint_changelog.yaml.template

```diff
@@ -0,0 +1,62 @@
+{{{{ .Header }}}}
+name: lint-changelog
+
+on:
+  pull_request:
+    paths:
+      - CHANGELOG.md
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: '0'
+    - name: Install dependencies
+      run: |
+        pip install parse-changelog
+    - name: Parse changelog to JSON
+      run: |
+        parse-changelog --changelog ./CHANGELOG.md > _changelog.json
+    - name: Print parsed changelog
+      run: |
+        jq . _changelog.json
+    - name: Check if introduction is present
+      run: |
+        jq -e 'has("introduction")' _changelog.json || (echo "::error file=CHANGELOG.md,line=1,col=1,endColumn=1::Missing introduction" && exit 1)
+    - name: Check if introduction title is valid
+      run: |
+        jq -e '.introduction | has("title")' _changelog.json || (echo "::error file=CHANGELOG.md,line=1,col=1,endColumn=1::Missing introduction title" && exit 1)
+        
+        TITLE=$(jq -r '.introduction.title' _changelog.json)
+        if [ -z "$TITLE" ]
+        then
+          echo "::error file=CHANGELOG.md,line=1,col=1::Empty introduction title"
+        fi
+    - name: Check if introduction content is valid
+      run: |
+        jq -e '.introduction | has("content")' _changelog.json || (echo "::error file=CHANGELOG.md,line=1,col=1::Missing introduction content" && exit 1)
+        
+        CONTENT=$(jq -r '.introduction.content' _changelog.json)
+        if [ -z "$CONTENT" ]
+        then
+          echo "::error file=CHANGELOG.md,line=1,col=1::Empty introduction content"
+        fi
+    - name: Check length
+      run: |
+        LENGTH=$(jq -r length _changelog.json)
+        if [[ "$LENGTH" -lt "2" ]]
+        then
+          echo "::error file=CHANGELOG.md,line=1,col=1::Changelog has not enough releases ($LENGTH), or the Markdown structure is incorrect."
+        else
+          echo "Found $LENGTH top-level items"
+        fi
+    - name: Check for unreleased changes
+      run: |
+        CHANGES=$(parse-changelog --changelog ./CHANGELOG.md --show-unreleased | tr -d '[:space:]')
+        if [ -z "$CHANGES" ]
+        then
+          echo "::warning file=CHANGELOG.md,line=1,col=1,endColumn=1::Please don't forget to add info about your changes to CHANGELOG.md"
+        fi
```

Comment:
Could we check for trailing whitespace, maybe? People sometimes introduce those, and they get trimmed away by reasonably-configured editors and that can be conflicting/annoying on rebase. I'd like to avoid that overall, even if it sounds minor.

Reply:
So basically we want strict Markdown linting on changelogs?

Reply:
Markdown allows extra whitespace, and even gives it special meaning (google "markdown two spaces"), but trailing whitespace should be avoided everywhere since some people configure their editors to trim them correctly (and everyone should). So it's not about Markdown linting, but I'd be in favor of running `markdownlint` against CHANGELOG.md if the findings are shown in GitHub PRs.